dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/JpaQueryUtils.java

```diff
@@ -341,8 +344,32 @@ public static String createSelectOrderExpression( @Nullable List<org.hisp.dhis.q
         };
     }
 
+    /**
+     *  Return SQL query for checking sharing access for given user
+     * @param sharingColumn sharing column reference
+     * @param user User for sharing checking
+     * @return SQL query
+     */
+    public static final String generateSQlQueryForSharingCheck( String sharingColumn, User user, String access )
+    {
+        List<Long> userGroupIds = getIdentifiers( user.getGroups() );
+
+        String groupsIds = CollectionUtils.isEmpty( userGroupIds ) ? null
+            : "{" + org.apache.commons.lang3.StringUtils.join( user.getGroups().stream().map( g -> g.getUid() )
+                .collect( Collectors.toList() ), "," ) + "}";
+
+        return  " ( " + sharingColumn + "->>'owner' is not null and " + sharingColumn + "->>'owner' = '" + user.getUid() + "') "
```

Comment:
Would it be possible to rewrite this using String.format, or similar to make it a little bit more readable?

Reply:
fixed